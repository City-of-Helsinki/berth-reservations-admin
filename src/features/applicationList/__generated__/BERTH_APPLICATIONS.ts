/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { ApplicationStatus, LeaseStatus } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL query operation: BERTH_APPLICATIONS
// ====================================================

export interface BERTH_APPLICATIONS_berthApplications_edges_node_customer {
  __typename: "ProfileNode";
  id: string;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_berthSwitch_reason {
  __typename: "BerthSwitchReasonType";
  title: string | null;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_berthSwitch {
  __typename: "BerthSwitchType";
  berthNumber: string;
  harbor: string;
  harborName: string;
  id: string;
  pier: string;
  reason: BERTH_APPLICATIONS_berthApplications_edges_node_berthSwitch_reason | null;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_lease_order {
  __typename: "OrderNode";
  id: string;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth_pier_properties_harbor_properties {
  __typename: "HarborProperties";
  name: string | null;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth_pier_properties_harbor {
  __typename: "HarborNode";
  id: string;
  properties: BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth_pier_properties_harbor_properties | null;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth_pier_properties {
  __typename: "PierProperties";
  identifier: string;
  harbor: BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth_pier_properties_harbor;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth_pier {
  __typename: "PierNode";
  properties: BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth_pier_properties | null;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth {
  __typename: "BerthNode";
  number: string;
  pier: BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth_pier;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_lease {
  __typename: "BerthLeaseNode";
  id: string;
  status: LeaseStatus;
  order: BERTH_APPLICATIONS_berthApplications_edges_node_lease_order | null;
  berth: BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_harborChoices {
  __typename: "HarborChoiceType";
  harbor: string;
  priority: number;
  harborName: string;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node {
  __typename: "BerthApplicationNode";
  id: string;
  firstName: string;
  lastName: string;
  customer: BERTH_APPLICATIONS_berthApplications_edges_node_customer | null;
  berthSwitch: BERTH_APPLICATIONS_berthApplications_edges_node_berthSwitch | null;
  email: string;
  createdAt: any;
  municipality: string;
  boatType: string | null;
  boatRegistrationNumber: string;
  boatWidth: number;
  boatLength: number;
  boatDraught: number | null;
  boatWeight: number | null;
  boatName: string;
  boatModel: string;
  accessibilityRequired: boolean;
  status: ApplicationStatus;
  lease: BERTH_APPLICATIONS_berthApplications_edges_node_lease | null;
  harborChoices: (BERTH_APPLICATIONS_berthApplications_edges_node_harborChoices | null)[] | null;
}

export interface BERTH_APPLICATIONS_berthApplications_edges {
  __typename: "BerthApplicationNodeEdge";
  node: BERTH_APPLICATIONS_berthApplications_edges_node | null;
}

export interface BERTH_APPLICATIONS_berthApplications {
  __typename: "BerthApplicationNodeConnection";
  count: number;
  edges: (BERTH_APPLICATIONS_berthApplications_edges | null)[];
}

export interface BERTH_APPLICATIONS_boatTypes {
  __typename: "BoatTypeType";
  id: string;
  name: string | null;
}

export interface BERTH_APPLICATIONS {
  berthApplications: BERTH_APPLICATIONS_berthApplications | null;
  boatTypes: BERTH_APPLICATIONS_boatTypes[] | null;
}

export interface BERTH_APPLICATIONSVariables {
  first: number;
  after?: string | null;
  switchApplications?: boolean | null;
  orderBy?: string | null;
  statuses?: (ApplicationStatus | null)[] | null;
}
