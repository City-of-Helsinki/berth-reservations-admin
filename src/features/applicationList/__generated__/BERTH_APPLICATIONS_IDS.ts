/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { ApplicationStatus } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL query operation: BERTH_APPLICATIONS_IDS
// ====================================================

export interface BERTH_APPLICATIONS_IDS_berthApplications_pageInfo {
  __typename: "PageInfo";
  hasNextPage: boolean;
  endCursor: string | null;
}

export interface BERTH_APPLICATIONS_IDS_berthApplications_edges_node {
  __typename: "BerthApplicationNode";
  id: string;
}

export interface BERTH_APPLICATIONS_IDS_berthApplications_edges {
  __typename: "BerthApplicationNodeEdge";
  node: BERTH_APPLICATIONS_IDS_berthApplications_edges_node | null;
}

export interface BERTH_APPLICATIONS_IDS_berthApplications {
  __typename: "BerthApplicationNodeConnection";
  count: number;
  pageInfo: BERTH_APPLICATIONS_IDS_berthApplications_pageInfo;
  edges: (BERTH_APPLICATIONS_IDS_berthApplications_edges | null)[];
}

export interface BERTH_APPLICATIONS_IDS {
  berthApplications: BERTH_APPLICATIONS_IDS_berthApplications | null;
}

export interface BERTH_APPLICATIONS_IDSVariables {
  first: number;
  after?: string | null;
  applicationCode?: boolean | null;
  switchApplications?: boolean | null;
  orderBy?: string | null;
  statuses?: (ApplicationStatus | null)[] | null;
  nameFilter?: string | null;
}
