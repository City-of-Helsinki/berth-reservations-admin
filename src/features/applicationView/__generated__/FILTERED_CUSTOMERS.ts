/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { OrganizationType } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL query operation: FILTERED_CUSTOMERS
// ====================================================

export interface FILTERED_CUSTOMERS_profiles_edges_node_primaryAddress {
  __typename: "AddressNode";
  address: string;
  postalCode: string;
  city: string;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_organization {
  __typename: "OrganizationNode";
  businessId: string;
  organizationType: OrganizationType;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties_harbor_properties {
  __typename: "HarborProperties";
  name: string | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties_harbor {
  __typename: "HarborNode";
  id: string;
  properties: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties_harbor_properties | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties {
  __typename: "PierProperties";
  identifier: string;
  harbor: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties_harbor;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier {
  __typename: "PierNode";
  properties: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth {
  __typename: "BerthNode";
  number: number;
  pier: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node {
  __typename: "BerthLeaseNode";
  berth: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges {
  __typename: "BerthLeaseNodeEdge";
  node: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases {
  __typename: "BerthLeaseNodeConnection";
  edges: (FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges | null)[];
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges_node_berthSwitch {
  __typename: "BerthSwitchType";
  harborName: string;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges_node {
  __typename: "BerthApplicationNode";
  berthSwitch: FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges_node_berthSwitch | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges {
  __typename: "BerthApplicationNodeEdge";
  node: FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges_node | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthApplications {
  __typename: "BerthApplicationNodeConnection";
  edges: (FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges | null)[];
}

export interface FILTERED_CUSTOMERS_profiles_edges_node {
  __typename: "ProfileNode";
  id: string;
  firstName: string;
  lastName: string;
  primaryAddress: FILTERED_CUSTOMERS_profiles_edges_node_primaryAddress | null;
  organization: FILTERED_CUSTOMERS_profiles_edges_node_organization | null;
  berthLeases: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases | null;
  berthApplications: FILTERED_CUSTOMERS_profiles_edges_node_berthApplications | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges {
  __typename: "ProfileNodeEdge";
  node: FILTERED_CUSTOMERS_profiles_edges_node | null;
}

export interface FILTERED_CUSTOMERS_profiles {
  __typename: "ProfileNodeConnection";
  count: number | null;
  edges: (FILTERED_CUSTOMERS_profiles_edges | null)[];
}

export interface FILTERED_CUSTOMERS {
  profiles: FILTERED_CUSTOMERS_profiles | null;
}

export interface FILTERED_CUSTOMERSVariables {
  first: number;
  after?: string | null;
  firstName?: string | null;
  lastName?: string | null;
  email?: string | null;
  address?: string | null;
  orderBy?: string | null;
}
