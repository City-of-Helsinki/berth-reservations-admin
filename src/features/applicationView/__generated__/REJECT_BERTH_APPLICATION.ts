/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { RejectBerthApplicationMutationInput } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: REJECT_BERTH_APPLICATION
// ====================================================

export interface REJECT_BERTH_APPLICATION_rejectBerthApplication {
  __typename: "RejectBerthApplicationMutationPayload";
  clientMutationId: string | null;
}

export interface REJECT_BERTH_APPLICATION {
  rejectBerthApplication: REJECT_BERTH_APPLICATION_rejectBerthApplication | null;
}

export interface REJECT_BERTH_APPLICATIONVariables {
  input: RejectBerthApplicationMutationInput;
}
