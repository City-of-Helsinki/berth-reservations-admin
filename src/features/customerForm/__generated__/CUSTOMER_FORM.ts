/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { CustomerGroup, InvoicingType, Language } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL query operation: CUSTOMER_FORM
// ====================================================

export interface CUSTOMER_FORM_profile_organization {
  __typename: "OrganizationNode";
  id: string;
  address: string;
  businessId: string;
  city: string;
  name: string;
  postalCode: string;
}

export interface CUSTOMER_FORM_profile_primaryAddress {
  __typename: "AddressNode";
  /**
   * The ID of the object
   */
  id: string;
  address: string;
  postalCode: string;
  city: string;
}

export interface CUSTOMER_FORM_profile_primaryEmail {
  __typename: "EmailNode";
  /**
   * The ID of the object
   */
  id: string;
  email: string;
}

export interface CUSTOMER_FORM_profile_primaryPhone {
  __typename: "PhoneNode";
  /**
   * The ID of the object
   */
  id: string;
  phone: string;
}

export interface CUSTOMER_FORM_profile {
  __typename: "ProfileNode";
  /**
   * The ID of the object
   */
  id: string;
  comment: string | null;
  firstName: string;
  lastName: string;
  customerGroup: CustomerGroup | null;
  invoicingType: InvoicingType | null;
  organization: CUSTOMER_FORM_profile_organization | null;
  /**
   * Convenience field for the address which is marked as primary.
   */
  primaryAddress: CUSTOMER_FORM_profile_primaryAddress | null;
  /**
   * Convenience field for the email which is marked as primary.
   */
  primaryEmail: CUSTOMER_FORM_profile_primaryEmail | null;
  /**
   * Convenience field for the phone which is marked as primary.
   */
  primaryPhone: CUSTOMER_FORM_profile_primaryPhone | null;
  language: Language | null;
}

export interface CUSTOMER_FORM {
  /**
   * Get profile by profile ID.
   * 
   * Requires `staff` credentials for the requester's service.The profile must have an active connection to the requester's service, otherwise it will not be returned.
   */
  profile: CUSTOMER_FORM_profile | null;
}

export interface CUSTOMER_FORMVariables {
  id: string;
}
