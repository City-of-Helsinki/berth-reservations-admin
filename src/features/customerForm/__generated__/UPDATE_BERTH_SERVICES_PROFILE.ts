/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { UpdateBerthServicesProfileMutationInput } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: UPDATE_BERTH_SERVICES_PROFILE
// ====================================================

export interface UPDATE_BERTH_SERVICES_PROFILE_updateBerthServicesProfile {
  __typename: "UpdateBerthServicesProfileMutationPayload";
  clientMutationId: string | null;
}

export interface UPDATE_BERTH_SERVICES_PROFILE {
  updateBerthServicesProfile: UPDATE_BERTH_SERVICES_PROFILE_updateBerthServicesProfile | null;
}

export interface UPDATE_BERTH_SERVICES_PROFILEVariables {
  input: UpdateBerthServicesProfileMutationInput;
}
