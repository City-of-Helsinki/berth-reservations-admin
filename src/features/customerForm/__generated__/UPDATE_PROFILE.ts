/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { UpdateProfileMutationInput } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: UPDATE_PROFILE
// ====================================================

export interface UPDATE_PROFILE_updateProfile {
  __typename: "UpdateProfileMutationPayload";
  clientMutationId: string | null;
}

export interface UPDATE_PROFILE {
  updateProfile: UPDATE_PROFILE_updateProfile | null;
}

export interface UPDATE_PROFILEVariables {
  input: UpdateProfileMutationInput;
}
