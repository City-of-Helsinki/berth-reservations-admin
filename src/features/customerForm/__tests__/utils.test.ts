import {
  createAddressAndOrganizationProperties,
  createEmailProperties,
  createPhoneProperties,
  createSensitivedataInputs,
  createUpdateInputs,
  getCustomerFormValues,
  getIdentifiers,
  mapCustomerGroupAsOrganizationType,
} from '../utils';
import {
  mockOrganizationCustomerFormIdentifiers as orgIds,
  mockOrganizationCustomerFormValues as orgFormValues,
  mockOrganizationCustomerProfile as orgProfile,
  mockPrivateCustomerFormIdentifiers as privateIds,
  mockPrivateCustomerFormValues as privateFormValues,
  mockPrivateCustomerProfile as privateProfile,
} from '../__fixtures__/mockData';
import { CustomerGroup } from '../../../@types/__generated__/globalTypes';

describe('utils', () => {
  describe('getCustomerFormValues', () => {
    it('maps a private customer correctly', () => {
      expect(getCustomerFormValues(privateProfile)).toMatchSnapshot();
    });

    it('maps a private customer with minimum data correctly', () => {
      expect(
        getCustomerFormValues({
          ...privateProfile,
          comment: null,
          primaryAddress: null,
          primaryPhone: null,
          primaryEmail: null,
          sensitivedata: null,
        })
      ).toMatchSnapshot();
    });

    it('maps an organization customer correctly', () => {
      expect(getCustomerFormValues(orgProfile)).toMatchSnapshot();
    });

    it('maps an organization customer with minimum data correctly', () => {
      expect(
        getCustomerFormValues({
          ...orgProfile,
          comment: null,
          primaryAddress: null,
          primaryPhone: null,
          primaryEmail: null,
        })
      ).toMatchSnapshot();
    });
  });

  describe('getIdentifiers', () => {
    it('maps a private customer correctly', () => {
      expect(getIdentifiers(privateProfile)).toMatchSnapshot();
    });

    it('maps an organization customer correctly', () => {
      expect(getIdentifiers(orgProfile)).toMatchSnapshot();
    });
  });

  describe('mapCustomerGroupAsOrganizationType', () => {
    it('should map each organization customer group to an organization type', () => {
      const orgCustomerGroups = Object.values(CustomerGroup).filter((value) => value !== CustomerGroup.PRIVATE);
      expect.assertions(orgCustomerGroups.length);

      orgCustomerGroups.forEach((customerGroup) => {
        expect(mapCustomerGroupAsOrganizationType(customerGroup)).toBeDefined();
      });
    });

    it('should map CustomerGroup.PRIVATE as "undefined"', () => {
      expect(mapCustomerGroupAsOrganizationType(CustomerGroup.PRIVATE)).toBeUndefined();
    });
  });

  describe('createEmailProperties', () => {
    it('should return "updateEmails" if email is not empty and previously existed', () => {
      expect(createEmailProperties('test@example.com', 'TEST-EMAIL')).toEqual({
        updateEmails: [
          {
            email: 'test@example.com',
            id: 'TEST-EMAIL',
            primary: true,
          },
        ],
      });
    });

    it('should return "addEmails" if email is not empty and did not previously exist', () => {
      expect(createEmailProperties('test@example.com', '')).toEqual({
        addEmails: [
          {
            email: 'test@example.com',
            emailType: 'NONE',
            primary: true,
          },
        ],
      });
    });

    it('should return "removeEmails" if email is empty and previously existed', () => {
      expect(createEmailProperties('', 'TEST-EMAIL')).toEqual({
        removeEmails: ['TEST-EMAIL'],
      });
    });

    it('should return an empty object if email is empty and did not previously exist', () => {
      expect(createEmailProperties('', '')).toEqual({});
    });
  });

  describe('createPhoneProperties', () => {
    it('should return "updatePhones" if phone is not empty and previously existed', () => {
      expect(createPhoneProperties('test@example.com', 'TEST-PHONE')).toEqual({
        updatePhones: [
          {
            phone: 'test@example.com',
            id: 'TEST-PHONE',
            primary: true,
          },
        ],
      });
    });

    it('should return "addPhones" if phone is not empty and did not previously exist', () => {
      expect(createPhoneProperties('test@example.com', '')).toEqual({
        addPhones: [
          {
            phone: 'test@example.com',
            phoneType: 'NONE',
            primary: true,
          },
        ],
      });
    });

    it('should return "removePhones" if phone is empty and previously existed', () => {
      expect(createPhoneProperties('', 'TEST-PHONE')).toEqual({
        removePhones: ['TEST-PHONE'],
      });
    });

    it('should return an empty object if phone is empty and did not previously exist', () => {
      expect(createPhoneProperties('', '')).toEqual({});
    });
  });

  describe('createAddressAndOrganizationProperties', () => {
    it('should handle switch from organization customer to private customer', () => {
      const [addressProperties, organizationProperties] = createAddressAndOrganizationProperties(
        privateFormValues,
        orgIds
      );

      expect(addressProperties.addAddresses).toBeDefined();
      expect(organizationProperties.deleteOrganization).toBeDefined();

      expect(addressProperties.removeAddresses).toBeUndefined();
      expect(addressProperties.updateAddresses).toBeUndefined();
      expect(organizationProperties.organization).toBeUndefined();
    });

    describe('should handle switch from private customer to organization customer', () => {
      test('with pre-existing address', () => {
        const [addressProperties, organizationProperties] = createAddressAndOrganizationProperties(
          orgFormValues,
          privateIds
        );

        expect(addressProperties.removeAddresses).toBeDefined();
        expect(organizationProperties.organization).toBeDefined();

        expect(addressProperties.addAddresses).toBeUndefined();
        expect(addressProperties.updateAddresses).toBeUndefined();
        expect(organizationProperties.deleteOrganization).toBeUndefined();
      });

      test('with no pre-existing address', () => {
        const [addressProperties, organizationProperties] = createAddressAndOrganizationProperties(orgFormValues, {
          ...privateIds,
          primaryAddressId: undefined,
        });

        expect(addressProperties.removeAddresses).toEqual([]);
        expect(organizationProperties.organization).toBeDefined();

        expect(addressProperties.addAddresses).toBeUndefined();
        expect(addressProperties.updateAddresses).toBeUndefined();
        expect(organizationProperties.deleteOrganization).toBeUndefined();
      });
    });

    describe('should handle updating a private customer', () => {
      test('with pre-existing address', () => {
        const [addressProperties, organizationProperties] = createAddressAndOrganizationProperties(
          privateFormValues,
          privateIds
        );

        expect(addressProperties.updateAddresses).toBeDefined();

        expect(addressProperties.addAddresses).toBeUndefined();
        expect(addressProperties.removeAddresses).toBeUndefined();
        expect(organizationProperties.deleteOrganization).toBeUndefined();
        expect(organizationProperties.organization).toBeUndefined();
      });

      test('with no pre-existing address', () => {
        const [addressProperties, organizationProperties] = createAddressAndOrganizationProperties(privateFormValues, {
          ...privateIds,
          primaryAddressId: undefined,
        });

        expect(addressProperties.addAddresses).toBeDefined();

        expect(addressProperties.removeAddresses).toBeUndefined();
        expect(addressProperties.updateAddresses).toBeUndefined();
        expect(organizationProperties.deleteOrganization).toBeUndefined();
        expect(organizationProperties.organization).toBeUndefined();
      });
    });

    it('should handle updating an organization customer', () => {
      const [addressProperties, organizationProperties] = createAddressAndOrganizationProperties(orgFormValues, orgIds);

      expect(organizationProperties.organization).toBeDefined();

      expect(addressProperties.addAddresses).toBeUndefined();
      expect(addressProperties.removeAddresses).toBeUndefined();
      expect(addressProperties.updateAddresses).toBeUndefined();
      expect(organizationProperties.deleteOrganization).toBeUndefined();
    });
  });

  describe('createSensitivedataInputs', () => {
    describe('for private customers', () => {
      it('should set sensitivedata.ssn if ssn is filled and sensitivedata exists', () => {
        expect(createSensitivedataInputs(CustomerGroup.PRIVATE, '000000-0000', true)).toEqual({
          sensitivedata: {
            ssn: '000000-0000',
          },
        });
      });

      it('should set sensitivedata.ssn if ssn is filled and sensitivedata does not exist', () => {
        expect(createSensitivedataInputs(CustomerGroup.PRIVATE, '000000-0000', false)).toEqual({
          sensitivedata: {
            ssn: '000000-0000',
          },
        });
      });

      it('should set empty sensitivedata.ssn if ssn is empty sensitivedata exists', () => {
        expect(createSensitivedataInputs(CustomerGroup.PRIVATE, '', true)).toEqual({
          sensitivedata: {
            ssn: '',
          },
        });
      });

      it('should not set sensitivedata if ssn is empty and sensitivedata does not exist', () => {
        expect(createSensitivedataInputs(CustomerGroup.PRIVATE, '', false)).toBeUndefined();
      });
    });

    describe('for organization customers', () => {
      it('should set empty sensitivedata.ssn if sensitivedata exists', () => {
        expect(createSensitivedataInputs(CustomerGroup.COMPANY, '000000-0000', true)).toEqual({
          sensitivedata: {
            ssn: '',
          },
        });
        expect(createSensitivedataInputs(CustomerGroup.COMPANY, '', true)).toEqual({
          sensitivedata: {
            ssn: '',
          },
        });
      });

      it('should not set sensitivedata if sensitivedata does not exist', () => {
        expect(createSensitivedataInputs(CustomerGroup.COMPANY, '000000-0000', false)).toBeUndefined();
        expect(createSensitivedataInputs(CustomerGroup.COMPANY, '', false)).toBeUndefined();
      });
    });
  });

  describe('createUpdateInputs', () => {
    it('should handle switch from organization customer to private customer', () => {
      expect(createUpdateInputs(privateFormValues, orgIds)).toMatchSnapshot();
    });

    it('should handle switch from private customer to organization customer', () => {
      expect(createUpdateInputs(orgFormValues, privateIds)).toMatchSnapshot();
    });

    it('should handle updating a private customer', () => {
      expect(createUpdateInputs(privateFormValues, privateIds)).toMatchSnapshot();
    });

    it('should handle updating an organization customer', () => {
      expect(createUpdateInputs(orgFormValues, orgIds)).toMatchSnapshot();
    });
  });
});
