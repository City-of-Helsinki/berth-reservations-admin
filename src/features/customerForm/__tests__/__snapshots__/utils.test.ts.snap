// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils createUpdateInputs should handle switch from organization customer to private customer 1`] = `
Array [
  Object {
    "profile": Object {
      "addAddresses": Array [
        Object {
          "address": "Mariankatu 2",
          "addressType": "NONE",
          "city": "Helsinki",
          "postalCode": "00170",
          "primary": true,
        },
      ],
      "firstName": "Pehr Evind",
      "id": "MOCK-PROFILE",
      "lastName": "Svinhufvud",
      "sensitivedata": null,
      "updateEmails": Array [
        Object {
          "email": "test@example.com",
          "id": "MOCK-EMAIL",
          "primary": true,
        },
      ],
      "updatePhones": Array [
        Object {
          "id": "MOCK-PHONE",
          "phone": "+358 00 000 0000",
          "primary": true,
        },
      ],
    },
    "serviceType": "BERTH",
  },
  Object {
    "comment": "",
    "deleteOrganization": true,
    "id": "MOCK-PROFILE",
  },
]
`;

exports[`utils createUpdateInputs should handle switch from private customer to organization customer 1`] = `
Array [
  Object {
    "profile": Object {
      "firstName": "Lauri Kristian",
      "id": "MOCK-PROFILE",
      "lastName": "Relander",
      "removeAddresses": Array [
        "MOCK-ADDRESS",
      ],
      "sensitivedata": null,
      "updateEmails": Array [
        Object {
          "email": "test@example.com",
          "id": "MOCK-EMAIL",
          "primary": true,
        },
      ],
      "updatePhones": Array [
        Object {
          "id": "MOCK-PHONE",
          "phone": "+358 00 000 0000",
          "primary": true,
        },
      ],
    },
    "serviceType": "BERTH",
  },
  Object {
    "comment": "",
    "id": "MOCK-PROFILE",
    "organization": Object {
      "address": "Mariankatu 2",
      "businessId": "1234567-8",
      "city": "Helsinki",
      "name": "Suomi Oy",
      "organizationType": "COMPANY",
      "postalCode": "00170",
    },
  },
]
`;

exports[`utils createUpdateInputs should handle updating a private customer 1`] = `
Array [
  Object {
    "profile": Object {
      "firstName": "Pehr Evind",
      "id": "MOCK-PROFILE",
      "lastName": "Svinhufvud",
      "sensitivedata": null,
      "updateAddresses": Array [
        Object {
          "address": "Mariankatu 2",
          "city": "Helsinki",
          "id": "MOCK-ADDRESS",
          "postalCode": "00170",
        },
      ],
      "updateEmails": Array [
        Object {
          "email": "test@example.com",
          "id": "MOCK-EMAIL",
          "primary": true,
        },
      ],
      "updatePhones": Array [
        Object {
          "id": "MOCK-PHONE",
          "phone": "+358 00 000 0000",
          "primary": true,
        },
      ],
    },
    "serviceType": "BERTH",
  },
  Object {
    "comment": "",
    "id": "MOCK-PROFILE",
  },
]
`;

exports[`utils createUpdateInputs should handle updating an organization customer 1`] = `
Array [
  Object {
    "profile": Object {
      "firstName": "Lauri Kristian",
      "id": "MOCK-PROFILE",
      "lastName": "Relander",
      "sensitivedata": null,
      "updateEmails": Array [
        Object {
          "email": "test@example.com",
          "id": "MOCK-EMAIL",
          "primary": true,
        },
      ],
      "updatePhones": Array [
        Object {
          "id": "MOCK-PHONE",
          "phone": "+358 00 000 0000",
          "primary": true,
        },
      ],
    },
    "serviceType": "BERTH",
  },
  Object {
    "comment": "",
    "id": "MOCK-PROFILE",
    "organization": Object {
      "address": "Mariankatu 2",
      "businessId": "1234567-8",
      "city": "Helsinki",
      "name": "Suomi Oy",
      "organizationType": "COMPANY",
      "postalCode": "00170",
    },
  },
]
`;

exports[`utils getCustomerFormValues maps a private customer correctly 1`] = `
Object {
  "address": "Mariankatu 2",
  "businessId": "",
  "city": "Helsinki",
  "comment": "",
  "customerGroup": "PRIVATE",
  "email": "test@example.com",
  "firstName": "Risto Heikki",
  "lastName": "Ryti",
  "organizationName": "",
  "phone": "+358 00 000 0000",
  "postalCode": "00170",
  "ssn": "0",
}
`;

exports[`utils getCustomerFormValues maps a private customer with minimum data correctly 1`] = `
Object {
  "address": "",
  "businessId": "",
  "city": "",
  "comment": "",
  "customerGroup": "PRIVATE",
  "email": "",
  "firstName": "Risto Heikki",
  "lastName": "Ryti",
  "organizationName": "",
  "phone": "",
  "postalCode": "",
  "ssn": "0",
}
`;

exports[`utils getCustomerFormValues maps an organization customer correctly 1`] = `
Object {
  "address": "Mariankatu 2",
  "businessId": "1234567-8",
  "city": "Helsinki",
  "comment": "",
  "customerGroup": "COMPANY",
  "email": "test@example.com",
  "firstName": "Kyösti",
  "lastName": "Kallio",
  "organizationName": "Suomi Oy",
  "phone": "+358 00 000 0000",
  "postalCode": "00170",
  "ssn": "0",
}
`;

exports[`utils getCustomerFormValues maps an organization customer with minimum data correctly 1`] = `
Object {
  "address": "Mariankatu 2",
  "businessId": "1234567-8",
  "city": "Helsinki",
  "comment": "",
  "customerGroup": "COMPANY",
  "email": "",
  "firstName": "Kyösti",
  "lastName": "Kallio",
  "organizationName": "Suomi Oy",
  "phone": "",
  "postalCode": "00170",
  "ssn": "0",
}
`;

exports[`utils getIdentifiers maps a private customer correctly 1`] = `
Object {
  "id": "MOCK-PROFILE",
  "initialCustomerGroup": "PRIVATE",
  "primaryAddressId": "MOCK-ADDRESS",
  "primaryEmailId": "MOCK-EMAIL",
  "primaryPhoneId": "MOCK-PHONE",
}
`;

exports[`utils getIdentifiers maps an organization customer correctly 1`] = `
Object {
  "id": "MOCK-PROFILE",
  "initialCustomerGroup": "COMPANY",
  "primaryAddressId": undefined,
  "primaryEmailId": "MOCK-EMAIL",
  "primaryPhoneId": "MOCK-PHONE",
}
`;
